echopop/live/live_data_loading.py

```diff
@@ -0,0 +1,928 @@
+        for filename in filenames
+        if get_file_haul_number(filename, biology_file_ids.get(key, "")) in common_hauls
+    ]
+
+    # Get bad files for DEBUG
+    non_filtered_filenames = [
+        filename
+        for key, filenames in dataset_dict.items()
+        for filename in filenames
+        if get_file_haul_number(filename, biology_file_ids.get(key, "")) not in common_hauls
+    ]
+    # ---- Create list
+    non_filtered_filenames_lst = "\n".join(non_filtered_filenames)
+    print(
+        f"The following files are parts of incomplete filesets: \n" f"{non_filtered_filenames_lst}"
+    )
+
+    # Return the curated filename list
+    return filtered_filenames
+
+
+def compile_filename_format(file_name_format: str):
+
+    # Create a copy of `file_name_format`
+    regex_pattern = file_name_format
+
+    # Iterate through the keys from `LIVE_FILE_FORMAT_MAP` to format a regex pattern
+    for key, value in LIVE_FILE_FORMAT_MAP.items():
+        regex_pattern = regex_pattern.replace(f"{{{key}}}", value["expression"])
+    # ---- Replace the `FILE_ID` tag
+    regex_pattern = re.sub(r"\{FILE_ID:(.+?)\}", r"(?P<FILE_ID>\1)", regex_pattern)
+
+    # Compile the regex pattern and return the output
+    return re.compile(regex_pattern)
+
+
+def read_biology_csv(file: Path, pattern: re.Pattern, config_map: dict, pandas_kwargs: dict = {}):
+
+    # Read in the `*.csv` file
+    df = pd.read_csv(file, usecols=list(config_map["dtypes"].keys()), storage_options=pandas_kwargs)
+
+    # Validate the dataframe
+    # ---- Check for any missing columns
+    missing_columns = [key for key in config_map["dtypes"].keys() if key not in df.columns]
+    # ---- Raise Error, if needed
+    if missing_columns:
+        raise ValueError(
+            f"The following columns are missing from [{file}]: {', '.join(missing_columns)}!"
+        )
+    # ---- Ensure the correct datatypes
+    df_validated = df.astype(config_map["dtypes"])
+    # ---- Replace column names and drop
+    df_validated = df_validated.rename(columns=config_map["names"])
+
+    # Get the substring components that can be added to the DataFrame
+    filename_substrings = re.findall(r"\{([^:}]+)(?::[^}]+)?}", pattern)
+    # ---- Create sub-list of columns that can be added to the DataFrame
+    valid_tags = list(set(["HAUL", "SPECIES_CODE"]).intersection(set(filename_substrings)))
+
+    # Compile the filename regular expression
+    compiled_regex = compile_filename_format(pattern)
+    # ---- Create the `Match` object that will be used to parse the string
+    match_obj = compiled_regex.search(file)
+
+    # Iterate through the filename-derived tags and add them to the DataFrame
+    for i in valid_tags:
+        matched_key = LIVE_FILE_FORMAT_MAP[i]
+        df_validated[matched_key["name"]] = matched_key["dtype"](match_obj.group(i))
+
+    # Return the resulting DataFrame
+    return df_validated
+
+
+def infer_datetime_format(timestamp_str: Union[int, str]):
```

Comment:
I wonder if the patterns you included here are already covered in pandas [guess_datetime_format](https://pandas.pydata.org/docs/dev/reference/api/pandas.tseries.api.guess_datetime_format.html)? Since these seem to be pretty standard ones. 

Also, do you need these because the files from FEAT have different datetime formats? For now this obviously works, but I feel we can communicate with Alicia to see if she could make those uniform.